packages/material-ui/src/Container/Container.js

```diff
@@ -30,39 +30,39 @@ export const styles = (theme) => ({
 
     if (value !== 0) {
       acc[theme.breakpoints.up(breakpoint)] = {
-        maxWidth: value,
+        maxWidth: `${value}${theme.breakpoints.unit}`,
       };
     }
     return acc;
   }, {}),
   /* Styles applied to the root element if `maxWidth="xs"`. */
   maxWidthXs: {
     [theme.breakpoints.up('xs')]: {
-      maxWidth: Math.max(theme.breakpoints.values.xs, 444),
+      maxWidth: Math.max(`${theme.breakpoints.values.xs}${theme.breakpoints.unit}`, 444),
```

Comment:
Hey! Not sure about this line, I think it is likely to break:
![image](https://user-images.githubusercontent.com/12865914/96892317-6f905a00-1481-11eb-97a8-bc28325256ce.png)
I guess this would work:
      ```maxWidth: Math.max(theme.breakpoints.values.xs, theme.breakpoints.unit.endsWith('em') ? 27.75 : 444),```

27.75 comes from dividing 444 / 16 (default font size in browsers).

Reply:
I am happy to push this changes and open a new PR if there's no other way now that it is merged 😄

Reply:
Thanks for double checking. It's definitely broken.

Reply:
This behavior is surprising. What is 444? We get the same issue on https://github.com/mui-org/material-ui/blob/3745c1e3a1f65d5a1c13b2d1c9db98bf69bbca33/packages/material-ui/src/Dialog/Dialog.js#L76

Reply:
What do you think about simply not handling the case, going back to the previous logic?

Reply:
We have a regression on https://next--material-ui.netlify.app/getting-started/templates/sign-in/.

Reply:
Hey @oliviertassinari not sure what 444 pixels means as I've only started using this library few days ago, I assume is kind of like a minimum value for Xs screens/phones maybe that was added there, also cause the default value for the Xs breakpoint is 0. That's why I left it without any changes when I submitted the PR. 😄 
Yes, I can see that it is breaking... 
![image](https://user-images.githubusercontent.com/12865914/96992700-28a67100-1522-11eb-9453-4f35c85a93ed.png)

I think my solution works:

**Xs breakpoint 0em:** 
![image](https://user-images.githubusercontent.com/12865914/96993435-38728500-1523-11eb-8db5-f608c179b790.png)
**Xs breakpoint > 27.5em (400px in default browser experience):**
![image](https://user-images.githubusercontent.com/12865914/96993547-5e982500-1523-11eb-85cd-1bf6232f9ffd.png)

**Xs breakpoint 0 (default value) no units defined defaults to pixels:**
![image](https://user-images.githubusercontent.com/12865914/96993743-a28b2a00-1523-11eb-8871-6e26f9b52f1c.png)
**Xs breakpoint > 444:**
![image](https://user-images.githubusercontent.com/12865914/96993936-e1b97b00-1523-11eb-8cfa-b60b8236331b.png)


I think it should work with rem(s) as well as the expression works with endsWith('em')

I'll open a PR later if you are happy with it 😃

Reply:
`em` is relative to the font-size of the parent. I restored the logic to its initial state in 36ef0b078c5ccebfd6bf2ed372e1c5f86ec34a57. I think that we should leave each developer to customize it to match its expected outcome.

Reply:
But perhaps when users have defined their theme breakpoints in the theme on ems for better accessibility and responsiveness like:
```
const theme = createMuiTheme({
  breakpoints: {
    unit: 'em',
    values: {
      xs: 30,
      sm: 37.5,
      md: 60,
      lg: 80,
      xl: 120,
    },
  },
```

It will "break" the layout as it will set the MaxWidth to 444px instead of 30em (which in default setting will look like 480px)
Yes, `em`is relative to the size of the parent, that's the point of setting up breakpoints and width values in `em` so that it can adapt to different font size that the user might need for accessibility. The solution proposed above would work in any of the ways and gives more flexibility to people working with `em`. I understand that this is a niche case but it still won't harm users that use pixels anyways😃

Reply:
`xs` is designed to be 0, IMHO, the existing of 444 in the first place is wrong

Reply:
Yes `xs` was designed to be 0 as it is the **smallest breakpoint**. But in the case of defining **containers maxWidth** as this case is a 0 won't be a good value I guess that's why the 444 is there. The fix proposed what it does is that it will set the value either to 444px or 27.5em/rem depending on the units being used, or if the breakpoint is different from the default and bigger than 444px/27.5em it will take it as maxWidth which is what the other maxWidth (sm,md,lg...) do...
So when using:
```
      <Container maxWidth={"xs"}>
        {children}
      </Container>
```
For the width of the container it will use by default 444px when setting the breakpoints with no units or with pixels. An will use 27.5em for it if they are set in ems...
If we just leave it as it is... It will default use 444px which works for pixels but it won't work _well_ for em/rem units...

Reply:
Still It is a codebase which I've recently started working with it and I might be missing something... I think is worth asking someone else in the MUI team to double check what we are discussing :)

Reply:
I see now that this bug (at least for sm, md, lg, xl) is showing up across the codebase in some other places... i.e: Dialog Component

Reply:
The case of the Dialog is even harder to handle, I doubt we can solve it without custom overrides.